mlflow/types/utils.py

```diff
@@ -130,6 +131,15 @@ def _infer_datatype(data: Any) -> Optional[Union[DataType, Array, Object, AnyTyp
         While empty lists are inferred as AnyType instead of None after the support of AnyType.
         e.g. [] -> AnyType, [[], []] -> Array(Any)
     """
+    if isinstance(data, pydantic.BaseModel):
+        # TODO: add link to documentation
+        raise MlflowException.invalid_parameter_value(
+            "Pydantic objects are not supported for model signature inference. "
```

Comment:
```suggestion
            "MLflow does not support inferring model signature from input examples with Pydantic objects. "
            "Instead, define your PythonModels' `predict` method to have a type hint and MLflow will "
            "infer the model signature from Pydantic type hints. e.g. "
            "`def predict(self, model_input: list[PydanticType])`"
```

nit: `Pydantic objects are not supported for model signature inference. ` is not accurate.